src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -597,12 +586,12 @@ PropertyList Visitor::visitTypesafe(Parser::PropertyListNotEmptyContext* ctx) {
 VarOrTerm Visitor::visitTypesafe(Parser::VerbContext* ctx) {
   if (ctx->varOrIri()) {
     return visitTypesafe(ctx->varOrIri());
-  }
-  if (ctx->getText() == "a") {
+  } else if (ctx->getText() == "a") {
     // Special keyword 'a'
     return GraphTerm{Iri{"<http://www.w3.org/1999/02/22-rdf-syntax-ns#type>"}};
+  } else {
+    AD_FAIL()  // Should be unreachable.
```

Comment:
Okay, consider this comment obsolete.
The other case where I thought this might help also has the special `a` predicate , so we can leave the `GraphTerm{Iri{` construct, as it makes stuff more explicit, as long as we haven't found a way to (simply and cleanly) make this a `visitAlternative`.

Maybe add a comment here "TODO<qup42, joka921> Is there a way to make this `visitAlternative` in the presence of the `a` case" and also add it in the other rule that also returns a `VarOrTerm` then you can close this comment for now, as this is not super blocking.